src/components/progressCircular/js/progressCircularDirective.js

```diff
@@ -0,0 +1,184 @@
+ * @param {number=} value In determinate mode, this number represents the percentage of the
+ *     circular progress. Default: 0
+ * @param {number=} md-diameter This specifies the diameter of the circular progress. The value
+ * should be a pixel-size value (eg '100'). If this attribute is
+ * not present then a default value of '50px' is assumed.
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-progress-circular md-mode="determinate" value="..."></md-progress-circular>
+ *
+ * <md-progress-circular md-mode="determinate" ng-value="..."></md-progress-circular>
+ *
+ * <md-progress-circular md-mode="determinate" value="..." md-diameter="100"></md-progress-circular>
+ *
+ * <md-progress-circular md-mode="indeterminate"></md-progress-circular>
+ * </hljs>
+ */
+
+angular
+  .module('material.components.progressCircular')
+  .directive('mdProgressCircular', ['$window', '$$rAF', '$mdProgressCircular', MdProgressCircularDirective]);
+
+function MdProgressCircularDirective($window, $$rAF, $mdProgressCircular) {
+  var DEGREE_IN_RADIANS = $window.Math.PI / 180;
+
+  return {
+    restrict: 'E',
+    scope: {
+      value: '@',
+      mdDiameter: '@'
+    },
+    template:
+      '<svg xmlns="http://www.w3.org/2000/svg">' +
+        '<path fill="none"/>' +
+      '</svg>',
+    compile: function(element){
+      element.attr('aria-valuemin', 0);
+      element.attr('aria-valuemax', 100);
+      element.attr('role', 'progressbar');
+
+      return MdProgressCircularLink;
+    }
+  };
+
+  function MdProgressCircularLink(scope, element) {
+    var svg = angular.element(element[0].querySelector('svg'));
+    var path = angular.element(element[0].querySelector('path'));
+    var lastAnimationId = 0;
+
+    path.attr('stroke-width', $mdProgressCircular.strokeWidth + 'px');
+
+    scope.$watchGroup(['value', 'mdDiameter'], function(newValue, oldValue) {
+      var id = ++lastAnimationId;
+      var startTime = new $window.Date();
+      var animateTo = clamp(newValue[0]);
+      var animateFrom = clamp(oldValue[0]);
+      var changeInValue = animateTo - animateFrom;
+
+      var animationDuration = $mdProgressCircular.animationDuration;
+      var diameter = $window.parseFloat(newValue[1]) || $mdProgressCircular.progressSize;
+      var pathDiameter = diameter - $mdProgressCircular.strokeWidth;
+
+      element.attr('aria-valuenow', animateTo);
+
+      svg.css({
+        width: diameter + 'px',
+        height: diameter + 'px'
+      });
+
+      // The viewBox has to be applied via setAttribute, because it is
+      // case-sensitive. If jQuery is included in the page, `.attr` lowercases
+      // all attribute names.
+      svg[0].setAttribute('viewBox', '0 0 ' + diameter + ' ' + diameter);
+
+      // No need to animate it if the values are the same
+      if(animateTo === animateFrom){
+        path.attr('d', getSvgArc(animateTo, diameter, pathDiameter));
+      }else{
```

Comment:
Little nitpicky, but can you space the else from curly braces?
See in the Google Code Conventions [here](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting)